pkg/services/automation/generator.go

```diff
@@ -410,6 +411,18 @@ func GenerateResourceName(url gitproviders.RepoURL) string {
 	return models.ConstrainResourceName(url.RepositoryName())
 }
 
+// GetRandomString generates a random unique string and appends it to a given string.
+func GetRandomString(s string) string {
+	data := "abcdefghijklmnopqrstuvwyz1234567890"
+
+	b := make([]byte, 5)
+	for i := range b {
+		b[i] = data[rand.Intn(len(data))]
+	}
+
+	return s + string(b)
+}
+
```

Comment:
It's still not limiting the length of the string that's being returned...which could lead to errors.

Reply:
@bigkevmcd just to make sure I understand correctly: since the `len` of the generated byte slice `b` will always be 5, do you mean that we should limit the input string `s`? What would you suggest as a limit?

Reply:
There is a limit on branch names (at least on GitHub)

https://stackoverflow.com/questions/60045157/what-is-the-maximum-length-of-a-github-branch-name

And you might be able to take some ideas from https://github.com/gitops-tools/pkg/blob/main/names/generator.go#L25